### DISCUSSION TITLE
How to successfully configure an Account Deletion flow?

### DISCUSSION THREAD

**Discussion by SirWobbyTheFirst** [goauthentik/authentik#11155] [General]
I'm currently stuck trying to build out a Delete your Account flow and stage in Authentik, I've got the flows and stages down but the only problem is that a user that is trying to delete their account needs to be super user to be able to delete the account, if they are not super user then they get a Request Denied error.

![image](https://github.com/user-attachments/assets/7e952d96-1454-4b77-857a-5a74ea872851)

When I log in as my test user Tommy, I'll go to the User Settings, click the Delete Account button, enter the password as per the flow requirements and then it should go to a user-delete stage but then spits out Request has been denied. If I then set the test user as a super user, I can delete the account but then the test user has super user rights and can access the Admin Panel which is of course, a terrible idea for security.

I looked into the Roles section, creating a role called Authentik Users and adding Authentik Core (Can Delete User) to the Assigned Global Permissions section, but I can't assign users to the role, instead they can only be assigned to groups, so I created a group called Authentik Users and put the test user in there and assigned it to the Authentik Users role but that does not seem to trickle down. I gotta admit, the documentation has been infuriating.

So can some please explain like I'm five, how someone is supposed to configure a way to allow users to delete their accounts if they wish to leave the service?

**Comment by Loapu** [comment]
The RBAC permissions are not fit for this purpose. Giving those would allow the user to delete someone via the admin interface.

I also have a simple user delete flow which works fine. Would you be willing to share how you setup your flow? Maybe you included a group binding in one of the stages?

**Reply by SirWobbyTheFirst** [reply]
Hi Loapu,

Apologies for the delay, I've attached a couple of screenshots of the flow and its setup, the flow setup is such that the user should be prompted for their password which works and I have a policy in place that stops users created via Active Directory from trying to delete themselves, however this policy does not affect the test user in question.

![image](https://github.com/user-attachments/assets/69ecaf2f-97d6-4658-9fd1-6be0720e2b11)
![image](https://github.com/user-attachments/assets/791fbe89-650e-4904-9423-96d21781f9cb)
![image](https://github.com/user-attachments/assets/90019eab-b6d9-40c3-8351-c1250fedbe7d)

**Reply by Loapu** [reply]
Hmm, this is odd. My flow looks nearly the same. I just deny the flow based on user group and instead of a password stage I have a simple consent stage. Would you mind showing me the policies and stage bindings?

**Reply by SirWobbyTheFirst** [reply]
Here are the screenshots you asked for. At least I'm assuming this is what you asked for.

![image](https://github.com/user-attachments/assets/35780d08-584c-4ea2-8ac3-729f1ec637e5)
![image](https://github.com/user-attachments/assets/0da41a54-ae46-4483-97e7-3e377a8f7590)
![image](https://github.com/user-attachments/assets/a7309383-0356-411f-8dff-d0015729a563)
![image](https://github.com/user-attachments/assets/ee89038e-2deb-4677-8246-f8912c9bde6f)
![image](https://github.com/user-attachments/assets/f6126152-4b8e-4b81-bb7e-3a686e95309b)
![image](https://github.com/user-attachments/assets/c564adc9-b0eb-4fb4-8d53-ec3f62a8a755)

Would you be able to export your flow please Loapu?

**Reply by SirWobbyTheFirst** [reply]
@Loapu - Hey dude, have you had a chance to take a look at the flow setup?

**Reply by Loapu** [reply]
Sorry for the late response, I forget to setup notifications for discussions. I will have a look at this tomorrow and will report back

**Reply by Loapu** [reply]
Alright, @SirWobbyTheFirst, after running into your error, can you please go into the admin panel and have a look at the event log? There should be an event titled “Policy exception” or something like that. If there is no such event, you may need to enter the authentik logs. If you use Docker, you should be able to attach yourself to the server container and look out for errors.

**Reply by SirWobbyTheFirst** [reply]
@Loapu - I've had a look at both the event log in Authentik and the logs on the container console and there is no error generated when the Delete Account error appears. Is there perhaps an option to enable a really verbose log? Alternatively, maybe there is a log file within the container you know of that may contain more information?

Note: The test user is set as an Internal user not an External user, would that have any affect? If the test user is set to External then they don't have access to the user settings page.

**Reply by SirWobbyTheFirst** [reply]
@Loapu - I've identified it, it's the Password prompt stage, I disabled the policy that stops AD Users from deleting their account and removed the Password prompt stage and then ran through the Delete Account URL and it immediately worked, the test user was removed.

Then did another test but with the AD Users policy enabled which also works, as soon as I add the Password Prompt stage back in, it fails. Problem is now, I wanted to have a way to confirm a user wants to delete their account before going ahead, this just goes straight ahead.

So is it possible to have a Password Prompt or a Consent Prompt to confirm this is what the user wants?

**Reply by Loapu** [reply]
I use a simple consent prompt and it works, maybe just try that?

**Reply by SirWobbyTheFirst** [reply]
@Loapu - I've added a simple consent prompt to the flow but it just displays a Continue button, is there any way to add a message stating "This is irreversible, are you sure you want to delete your account?" with a Yes or No button prompt, or maybe a Cancel button in addition to the Continue button?

This is what I've got at the moment:

![{F8AFC339-C0FF-461D-90F7-F2664A6F2D76}](https://github.com/user-attachments/assets/ddd97c70-9bb1-4fa1-8a58-4a2e4eaf450d)

**Reply by Loapu** [reply]
You can try to use the ak_message function in a policy before the consent stage

**Reply by SirWobbyTheFirst** [reply]
I added a policy where it just contained an ak_message() function call with a message but it does not seem to appear at the consent stage.

![{BE713819-3880-4E76-BF84-512983156695}](https://github.com/user-attachments/assets/6cce1a00-2f07-40cb-8503-5bb4a74f4fa6)
![{3D92EE04-8DD6-486B-837F-F40D4422D331}](https://github.com/user-attachments/assets/25636e49-7231-4e31-86b3-a06a6179141e)
![{B6339A2C-CEDE-4947-B74F-05E2BA3587E4}](https://github.com/user-attachments/assets/9ae9d869-79ea-46c2-8247-dc8a9e3abf27)
![{BB824EC0-86CE-479F-AFF4-391186F7AF03}](https://github.com/user-attachments/assets/da009979-6475-4544-ab21-fbb25a7a9630)
![{E6CC9CB5-D274-41FA-878A-0EE9965F29B8}](https://github.com/user-attachments/assets/9b8df55f-4c1f-49bc-bfc4-97fe741aa62f)

**Reply by Loapu** [reply]
I think you have to enable the checkbox for when flow is planned or attach the policy to a stage before the consent stage

**Reply by SirWobbyTheFirst** [reply]
Nope, it didn't work, I have settled for a needlessly complicated method where it displays a prompt with a dropdown box asking them to choose Yes or No, if they choose Yes, the account is deleted, if it is No, then it sits there and returns False.

Would be great to redirect back to the original page but the flow_plan variable and its redirect method seem to be houdini.

**Reply by SirWobbyTheFirst** [reply]
I gave in and just did a consent stage. If they delete their account, they delete their account, I no longer care, I am stressed and it should not be remotely this difficult to implement a simple yes or no prompt and then redirect if they choose no. Thank you for your time and patience Loapu, I wish I had that same level.

**Reply by Loapu** [reply]
You're welcome, authentik has some pretty huge weaknesses when it comes to custom stuff like custom texts, design and stuff like that.

**Comment by SirWobbyTheFirst** [comment]
I'm done.
